extensions/sql-migration/src/api/azure.ts

```diff
@@ -263,27 +263,27 @@ export async function startDatabaseMigration(account: azdata.Account, subscripti
 	};
 }
 
-export async function getDatabaseMigration(account: azdata.Account, subscription: Subscription, regionName: string, migrationId: string, sessionId: string): Promise<DatabaseMigration> {
+export async function getMigrationStatus(account: azdata.Account, subscription: Subscription, migration: DatabaseMigration, sessionId: string, asyncUrl: string): Promise<DatabaseMigration> {
 	const api = await getAzureCoreAPI();
-	const path = `${migrationId}?api-version=2020-09-01-preview`;
+	const migrationOperationId = getMigrationOperationId(migration, asyncUrl);
+	const path = `${migration.id}?migrationOperationId=${migrationOperationId}&$expand=MigrationStatusDetails&api-version=2020-09-01-preview`;
 	const response = await api.makeAzureRestRequest(account, subscription, path, azurecore.HttpRequestMethod.GET, undefined, true, undefined, getSessionIdHeader(sessionId));
 	if (response.errors.length > 0) {
-		if (response.response.status === 404 && response.response.data.error.code === 'ResourceDoesNotExist') {
-			throw new Error(response.response.data.error.code);
-		}
 		throw new Error(response.errors.toString());
 	}
 	return response.response.data;
 }
 
-export async function getMigrationStatus(account: azdata.Account, subscription: Subscription, migration: DatabaseMigration, sessionId: string): Promise<DatabaseMigration> {
-	const api = await getAzureCoreAPI();
-	const path = `${migration.id}?$expand=MigrationStatusDetails&api-version=2020-09-01-preview`;
-	const response = await api.makeAzureRestRequest(account, subscription, path, azurecore.HttpRequestMethod.GET, undefined, true, undefined, getSessionIdHeader(sessionId));
-	if (response.errors.length > 0) {
-		throw new Error(response.errors.toString());
+function getMigrationOperationId(migration: DatabaseMigration, asyncUrl: string): string {
+	// migrationOperationId may be undefined when provisioning has failed
+	// fall back to the operationId from the asyncUrl in the create migration response
+	if (migration.properties.migrationOperationId) {
+		return migration.properties.migrationOperationId;
 	}
-	return response.response.data;
+
+	return asyncUrl
```

Comment:
What do you use this for?
The function name is 'getMigrationOperationId'. If migrationOperationId is null, you are returning the guid from async url.
Do you then pass it to the GET migration request with migrationOperationId = <guid>?

Reply:
yes, that will returns the initial status of the create migration call in the event it does get past that.